### PR TITLE
The 514 patch

### PR DESCRIPTION
**Pull request by AffectedArc07**
fuck the break Im doing this before I perish to burnout

## What Does This PR Do
Updates the code to take advantage of 514 features and get everyone on the new client.

## Why It's Good For The Game
New version good.

## Changelog
N/A

### CODE REVIEW

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 #define TOPIC_SPAM_DELAY	2		//2 ticks is about 2/10ths of a second; it was 4 ticks, but that caused too many clicks to be lost due to lag
 #define UPLOAD_LIMIT		10485760	//Restricts client uploads to the server to 10MB //Boosted this thing. What's the worst that can happen?
-#define MIN_CLIENT_VERSION	513		// Minimum byond major version required to play.
+#define MIN_CLIENT_VERSION	514		// Minimum byond major version required to play.
 									//I would just like the code ready should it ever need to be used.
-#define SUGGESTED_CLIENT_VERSION	513		// only integers (e.g: 513, 514) are useful here. This is the part BEFORE the ".", IE 513 out of 513.1542
-#define SUGGESTED_CLIENT_BUILD	1542		// only integers (e.g: 1542, 1543) are useful here. This is the part AFTER the ".", IE 1542 out of 513.1542
+#define SUGGESTED_CLIENT_VERSION	514		// only integers (e.g: 513, 514) are useful here. This is the part BEFORE the ".", IE 513 out of 513.1542
+#define SUGGESTED_CLIENT_BUILD	1556		// only integers (e.g: 1542, 1543) are useful here. This is the part AFTER the ".", IE 1542 out of 513.1542
```
Comment (AffectedArc07) [inline code/modules/client/client_procs.dm:12]: This isnt needed. 513 clients can connect to 514 servers.

Comment (AffectedArc07) [reply code/modules/client/client_procs.dm:12]: image

Comment (hal9000PR) [reply code/modules/client/client_procs.dm:12]: surely it should still be good to inform people that they are running on an outdated byond version?

Comment (AffectedArc07) [reply code/modules/client/client_procs.dm:12]: Yeah, but I dont want to lock them out

Comment (S34NW) [reply code/modules/client/client_procs.dm:12]: 513 clients are experiencing frequent crashes it seems

File: code/_compile_options.dm [hunk 2]
```diff
@@ -22,12 +22,12 @@
 #define IS_MODE_COMPILED(MODE) (ispath(text2path("/datum/game_mode/"+(MODE))))
 
 //Update this whenever you need to take advantage of more recent byond features
-#define MIN_COMPILER_VERSION 513
-#define MIN_COMPILER_BUILD 1514
+#define MIN_COMPILER_VERSION 514
+#define MIN_COMPILER_BUILD 1554
```
Comment (AffectedArc07) [inline code/_compile_options.dm:26]: **NOT** `1556` because dreamchecker only emulates `1554`

File: _build_dependencies.sh [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-export SPACEMANDMM_TAG=suite-1.6
+export SPACEMANDMM_TAG=suite-1.7
 # For TGUI
 export NODE_VERSION=12
 # Byond Major
-export BYOND_MAJOR=513
+export BYOND_MAJOR=514
 # Byond Minor
-export BYOND_MINOR=1528
+export BYOND_MINOR=1556
```
Comment (AffectedArc07) [inline _build_dependencies.sh:9]:
```suggestion
export BYOND_MINOR=1566
```

File: code/modules/client/client_procs.dm [hunk 4]
```diff
@@ -8,8 +8,8 @@
 #define UPLOAD_LIMIT		10485760	//Restricts client uploads to the server to 10MB //Boosted this thing. What's the worst that can happen?
 #define MIN_CLIENT_VERSION	513		// Minimum byond major version required to play.
 									//I would just like the code ready should it ever need to be used.
-#define SUGGESTED_CLIENT_VERSION	513		// only integers (e.g: 513, 514) are useful here. This is the part BEFORE the ".", IE 513 out of 513.1542
-#define SUGGESTED_CLIENT_BUILD	1542		// only integers (e.g: 1542, 1543) are useful here. This is the part AFTER the ".", IE 1542 out of 513.1542
+#define SUGGESTED_CLIENT_VERSION	514		// only integers (e.g: 513, 514) are useful here. This is the part BEFORE the ".", IE 513 out of 513.1542
+#define SUGGESTED_CLIENT_BUILD	1556		// only integers (e.g: 1542, 1543) are useful here. This is the part AFTER the ".", IE 1542 out of 513.1542
```
Comment (AffectedArc07) [inline code/modules/client/client_procs.dm:12]:
```suggestion
#define SUGGESTED_CLIENT_BUILD	1566		// only integers (e.g: 1542, 1543) are useful here. This is the part AFTER the ".", IE 1542 out of 513.1542
```

**Pull request merged**